src/widget/button.rs

```diff
@@ -247,18 +258,29 @@ impl<'a, S> Widget for Button<'a, S>
         let widget::UpdateArgs { id, state, style, rect, ui, .. } = args;
         let Button { show, maybe_label, .. } = self;
 
-        let (color, times_clicked) = {
+        let (color, times_clicked,is_down) = {
             let input = ui.widget_input(id);
             let color = style.color(ui.theme());
+            let mut is_down = false;
             let color = input.mouse().map_or(color, |mouse| {
                 if mouse.buttons.left().is_down() {
-                    color.clicked()
+                    is_down = true;
+                    if let Some(clicked_color_destruct) = style.clicked_color {
+                        clicked_color_destruct
+                    } else {
+                        color.clicked()
+                    }
```

Comment:
When using an `if let` with an `else` branch, I normally opt for using `match` instead as it tends to be a little more concise for multiple branches:
```rust
match style.clicked_color {
    Some(clicked_color_destruct) => clicked_color_destruct,
    None => color.clicked(),
}
```
This is just personal preference, but just thought I'd mention it!